### DISCUSSION TITLE
Can SU2 Simulate Axisymmetric Case using Periodic Boundary Condition?

### DISCUSSION THREAD

**Discussion by mskim0401** [su2code/SU2#1318] [Q&A]
Hi.
I'm trying to run 2D axisymmetric cases, but it seems to have an issue. #1315 
So I'm trying to run in several different way. One of them is to use periodic boundary condition to solve axisymmetric problem.
I'm following the method CFL3D did. 
[CFL3D_ASJ](https://turbmodels.larc.nasa.gov/jetsubsonic_val.html)

Using 1 degree rotated mesh, I apply periodic boundary condition to both side walls.
I did several cases, but all cases diverge.

Boundary conditions in configuration file are below:
```
% -------------------- BOUNDARY CONDITION DEFINITION --------------------------%
%
% Euler wall boundary marker(s) (NONE = no marker)
% Implementation identical to MARKER_SYM.
MARKER_EULER= ( inviscid_wall )
%
% Navier-Stokes (no-slip), constant heat flux wall  marker(s) (NONE = no marker)
% Format: ( marker name, constant heat flux (J/m^2), ... )
MARKER_HEATFLUX= ( noslip_wall, 0.0, base, 0.0 )
%
% Far-field boundary marker(s) (NONE = no marker)
MARKER_FAR= ( farfield, inflow, outflow )
%
% Periodic boundary marker(s) (NONE = no marker)
% Format: ( periodic marker, donor marker, rotation_center_x, rotation_center_y,
% rotation_center_z, rotation_angle_x-axis, rotation_angle_y-axis,
% rotation_angle_z-axis, translation_x, translation_y, translation_z, ... )
MARKER_PERIODIC= ( periodic_02, periodic_01, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 0.0 )
```

Is there someone who succeed in this way?
I would appreciate it if someone tell me any idea.
Thank you.

**Comment by bigfooted** [comment]
Regarding periodic, maybe @TobiKattmann  can help. But if you have a 3D wedge, then probably symmetry boundary conditions are sufficient, unless you have a tangential velocity component (rotating flow).

**Reply by TobiKattmann** [reply]
Unfortunately I have not used rotational-periodicty yet and I am not aware right now of a testcase that uses that. I guess you also use just one element in rotation-direction for the primal mesh which might cause some issues as all nodes are boundary nodes and in some instances there is hard-coded numerical schemes used for that (because one does not have full set of neighbors...although theoretically the periodic neighbors are there but I am not sure if in that case we have a logic that prevents the hard-coded numerics).

I know it is tedious but to see whats working and what not here is how I would try to corner the problem.
1. Make a full revolution simulation (full cake) which should work and be ground truth. Tune the numerics and inital+boundary conditions a bit such that the case converges as fast as possible. Maybe also go with a low resolution mesh that converges because we're trying to figure out what works in general -> making it work with a suitable higher resolution mesh should then be easier.
2. Make a 90 degree wedge (or: greedy piece of cake :cake: ) with multiple cell layers in rotational direction (to prevent the mentioned numerics issues) and compute it with symmetry and periodic boundaries and see if that works and compares well to the full revolution
3. Then tackle the 1 cell width mesh you currently have (diet piece of cake :) ). If all others work then we might have an issue with the one cell layer in particular. (Although I made a 1 cell thick pipe slice with streamwise periodic boundary conditions in the past and that worked fine, albeit convergence was really slow)

**Reply by mskim0401** [reply]
Thank you all. 
I'm preparing to run both 1 and 2 steps (full 3D mesh and a 90 degree wedge). 
For the 90 degree wedge, I'll test several settings (symmetric BC or periodic BC with 90 degree to the side walls)

I hope this settings will do well.

Thank you very much.
